lib/IDE/SourceEntityWalker.cpp

```diff
@@ -321,7 +328,8 @@ std::pair<bool, Expr *> SemaAnnotator::walkToExprPre(Expr *E) {
     }
   }
 
-  if (!isa<InOutExpr>(E) &&
+  if (!IsImplicitDeclRefToExplicitDecl &&
+      !isa<InOutExpr>(E) &&
```

Comment:
Another option is to just add `!isa<DeclRefExpr>(E)` here, but I'm not sure of the implications of that, so opted for the safer option.

Reply:
I agree that checking for implicit references to explicit decls is the safe choice here.  There may be fully implicit cases that we want to handle, but my feeling is we're better off handling those on a case-by-case basis.